main.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/csv"
+	"encoding/hex"
+	"fmt"
+	"image"
+	"image/color"
+	"log"
+	"math"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	set "github.com/deckarep/golang-set"
+	"github.com/llgcode/draw2d/draw2dimg"
+)
+
+type LRTuple struct {
+	dir  float64
+	dist int
+}
+
+type Tuple struct {
+	rad float64
+	gx  int
+	gy  int
+}
+
+var ArrInput []LRTuple
+var xyInput []Tuple
+var iInput []Tuple
+var xyOutput []interface{}
+
+func printErr(err error) {
+	if err != nil {
+		log.Println(err.Error())
+	}
+}
+
+//parse and read textfile
+func readFile() {
+	file, err := os.Open("input.txt")
+	printErr(err)
+	defer file.Close()
+
+	reader := csv.NewReader(file)
+	result, err := reader.Read()
+	printErr(err)
+
+	log.Println(len(result))
+
+	for z := range result {
+		trimResult := strings.Trim(result[z], " ")
```

Comment:
Your `for z := range result {` is giving you an index, as you've noticed (since you're using result[z]).

Instead do `for index, value := range result {` to get the value out directly